### DISCUSSION TITLE
If the best solution don't improve after for example 120 seconds, could we terminate solver automatically ?

### DISCUSSION THREAD

**Discussion by C8504** [google/or-tools#2691] [CP-SAT questions] [answered]
Hi folks,

If the best solution don't improve after for example 120 seconds, could we terminate solver automatically by parameters in SAT solver? Although the total timelimit is 600 seconds.

John

**Comment by lperron** [comment, marked as answer]
Yes. But you have to do it yourself.  Create a solution callback. Receive the feasible solutions. Spawn another thread. Watch the time of the last solution, watch the current time, and call callback.StopSearch() if needed.

**Reply by C8504** [reply]
Got it, thanks Laurent.
